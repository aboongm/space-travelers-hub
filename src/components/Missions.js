import React from 'react';
import Mission from './Mission';

export default function Missions() {
  return (
    <table className="my-5">
      <tbody className="p-5">
        <tr className="border">
          <th className="border">Mission</th>
          <th className="border">Description</th>
          <th className="border">Status</th>
          <th className="border"> </th>
        </tr>
        <Mission />
      </tbody>
    </table>
  );
}
