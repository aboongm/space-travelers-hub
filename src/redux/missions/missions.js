import React from 'react';

export default function missions() {
  return (
    <div>missions</div>
  );
}
