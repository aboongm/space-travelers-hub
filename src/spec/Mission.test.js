import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Missions from '../components/Missions';
import store from '../redux/configureStore';
import { joinMissionAction, leaveMissionAction, fetchMissionApiAction } from '../redux/missions/missions';

describe('Missions tests', () => {
  it('render test', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should return an action with type GET_MISSIONS', () => {
    const data = [
      {
        id: 1,
        name: 'mission',
        description: 'description',
      },
    ];
    const action = fetchMissionApiAction(data);
    expect(action.type).toBe('GET_MISSIONS');
  });

  it('should return an action with type JOIN_MISSION', () => {
    const id = 1;
    const action = joinMissionAction(id);
    expect(action.type).toBe('JOIN_MISSION');
  });

  it('should should return an action with type LEAVE_MISSION', () => {
    const id = 1;
    const action = leaveMissionAction(id);
    expect(action.type).toBe('LEAVE_MISSION');
  });
});
