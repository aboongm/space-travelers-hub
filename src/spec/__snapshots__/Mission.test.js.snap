// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions tests render test 1`] = `
<div
  className="px-3 container"
>
  <table
    className="my-5 "
  >
    <tbody
      className="p-5"
    >
      <tr
        className="border"
      >
        <th
          className="border"
        >
          Mission
        </th>
        <th
          className="border"
        >
          Description
        </th>
        <th
          className="border"
        >
          Status
        </th>
        <th
          className="border"
        >
           
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
